code/__HELPERS/AStar.dm

```diff
@@ -61,31 +60,104 @@ Actual Adjacent procs :
 /datum/pathnode/proc/calc_f()
 	f = g + h
 
+
+
 //////////////////////
 //A* procs
 //////////////////////
 
 //the weighting function, used in the A* algorithm
-/proc/PathWeightCompare(datum/pathnode/a, datum/pathnode/b)
+/proc/PathWeightCompare(datum/jpsnode/a, datum/jpsnode/b)
```

Comment:
If you can turn these auxiliary functions into defines, that'd be positive.
for example, `#define PATH_WEIGHT_COMPARE(node_a, node_b) (node_a.f - node_b.f)